### PR TITLE
Add redis checks to disclosures

### PR DESCRIPTION
**Pull request by flooie**
PR - Fixes celery duplication problem in the financial disclosure importer.

Uses redis to handle race conditions with celery in financial disclosure extraction. 
Moves has_been_extracted to disclosure.tasks.

### CODE REVIEW

File: cl/disclosures/tasks.py [hunk 1]
```diff
@@ -476,3 +523,27 @@ def import_disclosure(self, data: Dict[str, Union[str, int, list]]) -> None:
 
     # Save PDF content
     save_disclosure(extracted_data=content, disclosure=disclosure)
+    # Remove disclosure ID in redis for completed disclosure
+    interface.delete(disclosure_key)
+
+
+def has_been_extracted(data: Dict[str, Union[str, int, list]]) -> bool:
```
Comment (mlissner) [inline cl/disclosures/tasks.py:530]: I think I'd put this in `cl.disclosures.utils`. It's not a task and doing this is a bit weird:

`from cl.disclosures.tasks import has_been_extracted`

Reading that, I'd think that `has_been_extracted` would be a task, not a utility?

File: cl/disclosures/tasks.py [hunk 2]
```diff
@@ -402,10 +435,22 @@ def import_disclosure(self, data: Dict[str, Union[str, int, list]]) -> None:
...
-
     from cl.disclosures.models import FinancialDisclosure
     from cl.people_db.models import Person
 
+    # Check download_filepath to see if it has been processed before.
+    if has_been_extracted(data):
+        logger.info(f"Document already extracted and saved: {data['id']}.")
+        return
+
+    interface = make_redis_interface("DISCLOSURES")
```
Comment (mlissner) [inline cl/disclosures/tasks.py:446]: use `CACHE` here, not `DISCLOSURES`. We don't need a full redis DB just for this (and if we did, we'd have to add it in our settings.)

File: cl/disclosures/tasks.py [hunk 3]
```diff
@@ -476,3 +523,27 @@ def import_disclosure(self, data: Dict[str, Union[str, int, list]]) -> None:
 
     # Save PDF content
     save_disclosure(extracted_data=content, disclosure=disclosure)
+    # Remove disclosure ID in redis for completed disclosure
+    interface.delete(disclosure_key)
```
Comment (mlissner) [inline cl/disclosures/tasks.py:527]: Do we need to delete this key on line 522 as well?

**Review by mlissner — CHANGES_REQUESTED** [review]
Looks about right. A few little things.

File: cl/disclosures/tasks.py (hunk 1 above)
Comment (flooie) [reply cl/disclosures/tasks.py:530]: that makes lots of sense - I had something gnawing about this

File: cl/disclosures/tasks.py (hunk 3 above)
Comment (flooie) [reply cl/disclosures/tasks.py:527]: Yes

File: cl/disclosures/tasks.py (hunk 1 above)
Comment (flooie) [reply cl/disclosures/tasks.py:530]: I'll add has_been_pdfed as well

**Pull request merged**